src/app/parcels/batch/BatchDataRow.ts

```diff
@@ -0,0 +1,377 @@
+    dietaryRequirements: BooleanGroup;
+    feminineProducts: BooleanGroup;
+    babyProducts: boolean | null;
+    nappySize: string | null;
+    petFood: BooleanGroup;
+    otherItems: BooleanGroup;
+    deliveryInstructions: string | null;
+    extraInformation: string | null;
+    attentionFlag: boolean;
+    signpostingCall: boolean;
+    notes: string | null;
+}
+
+interface OverrideParcel {
+    [key: string]: string | null | CollectionInfo;
+    voucherNumber: string | null;
+    packingDate: string | null;
+    packingSlot: string | null;
+    shippingMethod: string | null;
+    collectionInfo: CollectionInfo | null;
+}
+
+interface BatchParcel {
+    [key: string]: string | null | CollectionInfo;
+    voucherNumber: string | null;
+    packingDate: string;
+    packingSlot: string;
+    shippingMethod: string;
+    collectionInfo: CollectionInfo | null;
+}
+
+interface OverrideData {
+    client: OverrideClient;
+    parcel: OverrideParcel;
+}
+
+interface BatchEditData {
+    client: BatchClient;
+    clientReadOnly: boolean;
+    parcel: BatchParcel | null;
+}
+interface OverrideDataRow {
+    data: OverrideData;
+}
+
+interface BatchDataRow {
+    id: number;
+    clientId: string | null;
+    data: BatchEditData | null;
+}
+
+export interface BatchTableDataState {
+    overrideDataRow: OverrideDataRow;
+    batchDataRows: BatchDataRow[];
+}
+
+export interface BatchActionType {
+    type: "update_cell" | "add_row" | "delete_row" | "override_column" | "use_existing_client";
+    payload?: BatchActionPayload;
+}
+
+interface BatchActionPayload {
+    rowId?: number;
+    newRow?: BatchDataRow;
+    newOverrideRow?: OverrideDataRow;
+    existingClientId?: string;
+}
+
+const getOverridenFields = (allFields: OverrideClient | OverrideParcel): string[] => {
+    return Object.entries(allFields)
+        .filter(([_, value]) => value)
+        .reduce((acc, [key, _]) => {
+            return [...acc, key];
+        }, [] as string[]);
+};
+
+function createBooleanGroupFromStrings(strings: string[] | null): BooleanGroup {
+    const result: BooleanGroup = {};
+    if (strings) {
+        strings.forEach((str) => {
+            result[str] = true;
+        });
+    }
+    return result;
+}
+
+const getNappySize = (info: string | null): string | null => {
+    if (info) {
+        const match = info.match(/Nappy Size:\s*(\d+)/);
+        if (match) {
+            return match[1];
+        }
+    }
+    return null;
+};
+
+const getChildrenAndAdults = async (
+    familyId: string
+): Promise<{ adults: Person[]; children: Person[] }> => {
+    const { data, error } = await getFamilySupabaseCall(familyId);
+
+    const adults: Person[] = [];
+    const children: Person[] = [];
+
+    const currentYear = new Date().getFullYear();
+    const currentMonth = new Date().getMonth();
+
+    if (error) {
+        logErrorReturnLogId("Error with fetch: family_members", { error: error });
+        return { adults: [], children: [] };
```

Comment:
What exactly happens at this point when the empty arrays are returned? The error is logged but is there any indication that an error occurred or does the adult, child info just show up empty? Might be outside scope of ticket but thought I'd bring it up.